mbq/metrics/__init__.py

```diff
@@ -13,31 +13,32 @@
 logger = logging.getLogger('mbq.metrics')
 
 _initialized = False
+_namespace = None
+_constant_tags = {}
 _statsd = utils.NullStatsd()
 
 
 def init(namespace=None, constant_tags=None):
-    global _initialized, _statsd
+    global _initialized, _namespace, _constant_tags, _statsd
     if _initialized:
         logger.warning('mbq.metrics already initialized. Ignoring re-init.')
         return
 
     if constant_tags:
-        constant_tags = utils.tag_dict_to_list(constant_tags)
+        _constant_tags = constant_tags
 
+    _namespace = namespace
     _statsd = datadog.DogStatsd(
-        namespace=namespace,
-        constant_tags=constant_tags,
         use_default_route=True,  # assumption: code is running in a container
     )
     _initialized = True
 
 
 class Collector(object):
-    def __init__(self, prefix=None, tags=None, statsd=None):
+    def __init__(self, prefix=None, tags=None, namespace=None):
         self.prefix = prefix
-        self.tags = tags or {}
-        self._statsd = statsd
+        self.tags = tags if tags else {}
```

Comment:
Need to pull in the constant tags here. See note about using a property.

Reply:
why do we need to pull in the constant tags here?  I decided to pull in the constant tags in the `_combine_metrics` function instead.